javascript/signin.js

```diff
@@ -0,0 +1,122 @@
+    clearErrorStyle(emailInput);
+
+    // 기존 에러 메세지 삭제
+    if (errorTextEmail.parentNode === emailInput.parentNode) {
+      emailInput.parentNode.removeChild(errorTextEmail);
+    }
+    // 이메일 미입력 에러
+    if (email == '') {
+      showError(errorTextEmail, '이메일을 입력해 주세요.', emailInput);
+      return false;
+    }
+    // 이메일 형식 확인
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    if (!emailRegex.test(email)) {
+      showError(errorTextEmail, '올바른 이메일 주소가 아닙니다.', emailInput);
+      return false;
+    }
+    return true;
+  }
+
+  // 비번 확인 함수
+  function passwordCheck() {
+    const password = passwordInput.value.trim();
+    clearErrorStyle(passwordInput);
+    // 기존 에러 메세지 삭제
+    if (errorTextPassword.parentNode === passwordInput.parentNode) {
+      passwordInput.parentNode.removeChild(errorTextPassword);
+    }
+    // 비밀번호 미입력 에러
+    if (password == '') {
+      showError(errorTextPassword, '비밀번호를 입력해 주세요.', passwordInput);
+      return false;
+    }
+    return true;
+  }
+
+  // 에러 스타일 빼기
+  function clearErrorStyle(inputElement) {
+    const parentContainer = inputElement.parentNode;
+    const errorElement = parentContainer.querySelector('.error-message');
+
+    parentContainer.classList.remove('error');
+    if (errorElement == true) {
+      parentContainer.removeChild(errorTextEmail);
+    }
+  }
+
+  // 에러 메세지 띄우기
+  function showError(errorText, message, inputElement) {
+    errorText.textContent = message;
+    errorText.classList.add('error-message');
+    inputElement.parentNode.appendChild(errorText);
+    inputElement.parentNode.classList.add('error');
+  }
+
+  // 눈 아이콘 변경하기
+  function togglePassword() {
+    const coverEye = '../image/icon/eye_closed.svg';
+    const openEye = '../image/icon/eye_open.svg';
+    if (isPasswordCovered) {
+      passwordCover.src = coverEye;
+    } else {
+      passwordCover.src = openEye;
+    }
+    isPasswordCovered = !isPasswordCovered;
+  }
+
+  function showPassword() {
+    if (isPasswordCovered) {
+      passwordInput.type = 'password';
+    } else {
+      passwordInput.type = 'text';
+    }
+  }
+
+  //////////////// 함수 사용////////////////////
+
+  // 이메일 이벤트 리스너 부여
+  emailInput.addEventListener('focusout', function () {
+    emailCheck();
+  });
+
+  // 비번 이벤트 리스너 부여
+  passwordInput.addEventListener('focusout', function () {
+    passwordCheck();
+  });
+
+  // 로그인 버튼 입력 시 이메일 비번 확인 후 이동
+  form.addEventListener('submit', function (event) {
+    const codeit = { email: 'test@codeit.com', password: 'codeit101' };
+    const email = emailInput.value.trim();
+    const password = passwordInput.value.trim();
+
+    if (email === codeit.email && password === codeit.password) {
+      event.preventDefault();
+      window.location.href = '../html/folder.html';
+    } else {
+      showError(errorTextEmail, '이메일을 확인해 주세요.', emailInput);
+      showError(errorTextPassword, '비밀번호를 확인해 주세요.', passwordInput);
+      event.preventDefault();
+    }

```

Comment:
if-else 문을 불가피하게 사용해야하는 경우를 제외하고는 사용하지 않는 것이 가독성 측면에서 더 좋습니다. else 문 대신 [early return](https://devscb.tistory.com/130)을 사용해보시면 좋습니다.

```suggestion
    if (email === codeit.email && password === codeit.password) {
      event.preventDefault();
      window.location.href = '../html/folder.html';
      return;
    }
     showError(errorTextEmail, '이메일을 확인해 주세요.', emailInput);
     showError(errorTextPassword, '비밀번호를 확인해 주세요.', passwordInput);
     event.preventDefault();
```

얼리 리턴은 유명한 리팩토링 기법인데요, 
if문을 사용하여 indent(코드 들여쓰기)가 깊어지면 가독성 측면에서 좋지 않기도 하고, 미세하게 코드 성능도 안좋기도 합니다. 그래서 이럴 때 빠른 return 문을 사용해서 코드를 개선해 볼 수 있어요.

Reply:
코드는 가능한 가볍고 가독성 좋게 만드는 편이 좋다는 뜻이군요! 수정했습니다~! c97a1e1b40c7e37083634ef31c99d77908939a79